Sources/ClientRuntime/EventStream/Message.swift

```diff
@@ -22,3 +22,52 @@ extension EventStream {
         }
     }
 }
+
+extension EventStream.Message: CustomDebugStringConvertible {
+    public var debugDescription: String {
+        let headers = self.headers.map { "\($0.name): \($0.value)" }.joined(separator: "\n")
+        return """
+        EventStream.Message(
+            headers: [
+            \(headers)
+            ],
+            payload: \(String(data: payload, encoding: .utf8) ?? "<invalid>"
+        )
+        """
+    }
+}
+
+extension Array where Element == EventStream.Header {
+    public func toHeaders() -> Headers {
+        var headers = Headers()
+        for header in self {
+            switch header.value {
+            case .bool(let value):
+                headers.add(name: header.name, value: value.description)
+            case .byte(let value):
+                headers.add(name: header.name, value: value.description)
+            case .int16(let value):
+                headers.add(name: header.name, value: value.description)
+            case .int32(let value):
+                headers.add(name: header.name, value: value.description)
+            case .int64(let value):
+                headers.add(name: header.name, value: value.description)
+            case .data(let value):
+                headers.add(name: header.name, value: value.base64EncodedString())
+            case .string(let value):
+                headers.add(name: header.name, value: value)
+            case .timestamp(let value):
+                headers.add(name: header.name, value: value.description)
+            case .uuid(let value):
+                headers.add(name: header.name, value: value.description)
+            }
+        }
+        return headers
+    }
+}
+
+extension Array where Element == EventStream.Header {
+    public func value(name: String) -> EventStream.HeaderValue? {
```

Comment:
Nit: IMO this function name is a bit too generic and depends too much on the array instance name to reason about the code. Can we add something header specific to the name?

```suggestion
    public func value(forHeader name: String) -> EventStream.HeaderValue? {
```

Reply:
actually name is right way to call the "key"

https://www.rfc-editor.org/rfc/rfc9110.html#fields